packages/itwin/ec3-widget/src/common/UseEC3WidgetLocalization.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useMemo } from "react";
+import { EC3WidgetUI } from "../EC3WidgetUI";
+
+export interface useEC3WidgetLocalizationResult {
+  stepOneTitle: string;
+  stepTwoTitle: string;
+  stepThreeTitle: string;
+  templateName: string;
+  templateDescription: string;
+  reportSelection: string;
+  reportSelectionPlaceholderLoading: string;
+  reportSelectionPlaceholderSelect: string;
+  assemblyName: string;
+  reportTable: string;
+  element: string;
+  elementPlaceholderSelect: string;
+  elementPlaceholderLoading: string;
+  selectReportTableFirstPlaceholder: string;
+  reportTablesPlaceholderLoading: string;
+  reportTablesPlaceholderSelect: string;
+  elementQuantity: string;
+  elementQuantityPlaceholderSelect: string;
+  materials: string;
+  materialsSelectPlaceholder: string;
+  backButton: string;
+  nextButton: string;
+  cancelButton: string;
+  savingFailedToasterMsg: string;
+  unAuthorisedUserMsg: string;
+  addNewAssembly: string;
+  selectionSummary: string;
+  saveButton: string;
+  noTemplateMsg: string;
+  requiredFieldNotice: string;
+  createTemplate: string;
+}
+
+/** @internal */
+export const compareLocalizedString = (defaultStrings: any, overriddenStrings: any) => {
```

Comment:
Usually comparison functions for objects returns an integer that indicates their relative position in the sort order. I don't understand what this function does from the name alone.

Reply:
changed the approach as discussed on call